cpp/ql/test/library-tests/dataflow/taint-tests/vector.cpp

```diff
@@ -529,6 +529,6 @@ void test_vector_iterator() {
 		sink(*it);
 		it += source();
 		sink(*it); // $ ast,ir
-		sink(vs[1]);
+		sink(vs[1]); // $ SPURIOUS: ir
```

Comment:
Where is this coming from?

Reply:
This one is super annoying. What happens is:
1. There's a write to the iterator at `it += source();`
2. This write propagates to `it` on the next line at `sink(*it);`
3. There's a taint step from `it` to `*it`
4. The `*it` is seen as a use of `vs` (because of the line `it = vs.begin();`)
5. There's use-use flow from `*it` (which is a use of `vs`) to `vs` on the line at `sink(vs[1]);`
6. There's a taint step from `vs` to `vs[1]`

Individually all of these steps make sense, but the final result is that we have taint flow from the write to the iterator to the use of the underlying container 😭. This only happens in taint configurations, though (since it requires a taint step to go from `it` to `*it`).